src/simulation.hpp

```diff
@@ -576,6 +578,33 @@ template <typename problem_t> void AMRSimulation<problem_t>::setInitialCondition
 		ReadCheckpointFile();
 	}
 
+#if AMREX_SPACEDIM == 3
+	// set up elliptic solve object
+	amrex::OpenBCSolver poissonSolver(Geom(0, finest_level), boxArray(0, finest_level), DistributionMap(0, finest_level));
```

Comment:
We should avoid code duplication by putting this code in its own function, and calling it from here and from `ellipticSolveAllLevels`.

Reply:
what do you want to call this function?

Reply:
@BenWibking